### ISSUE TITLE
Provide a helper to generate role-limited kubeconfig files

### ISSUE THREAD

**Issue by jkinkead** [pulumi/pulumi-eks#348]
### Problem description

Today, `eks.Cluster` [creates a kubeconfig file](https://github.com/pulumi/pulumi-eks/blob/951a1da6a398788a95e0ec7c23428ea91c2b1bd0/nodejs/eks/cluster.ts#L397-L408) automatically that either [embeds the role ARN used to create the cluster](https://github.com/pulumi/pulumi-eks/blob/951a1da6a398788a95e0ec7c23428ea91c2b1bd0/nodejs/eks/cluster.ts#L126-L128), or else falls back on whatever the default role happens to be being run at the time of cluster creation.

Typically, users will want to provide an ARN, so that the provider (and kubeconfig) are consistent regardless of who is running the Pulumi program. However, this is problematic, since this role will have `cluster:admin` permissions in Kubernetes, which is not desirable - and furthermore precludes things like having per-namespace Pulumi programs for multiple teams sharing a cluster, without granting all programs admin-level permissions.

What I'd like is for a method on `eks.Cluster` that takes a role ARN, and returns a valid `kubeconfig` file using that ARN - but with the cluster name, endpoint, and CA data already embedded.

Basically, something like:

```typescript
getKubeconfig(roleArn: string) {
  // Fudging Output mapping here.
  return generateKubeconfig(
    this.eksCluster.name,
    this.eksCluster.endpoint,
    this.eksCluster.certificateAuthority,
    roleArn);
}
```
